docs/topics/periodic-stats.rst

```diff
@@ -0,0 +1,298 @@
+    ...
+    """
+
+To export only stats when a spider is finished use the ``only_export_on_close`` parameter::
+
+    PERIODIC_STATS_OBSERVERS = [
+        stats.Observer(key='downloader/request_count', export_key='requests', only_export_on_close=True),
+    ]
+
+    """
+    Generated stats:
+    {}                  # for t=1
+    {}                  # for t=2
+    {}                  # for t=3
+    {}                  # for t=4
+    ...
+    {'requests': 2759}  # on spider_close
+    """
+
+Regular expressions can be used to group/filter data with the ``use_re_key`` parameter:::
+
+    PERIODIC_STATS_OBSERVERS = [
+        stats.Observer(key='downloader/response_status_count/2..', use_re_key=True, export_key='2xx'),
+        stats.Observer(key='downloader/response_status_count/3..', use_re_key=True, export_key='3xx'),
+        stats.Observer(key='downloader/response_status_count/4..', use_re_key=True, export_key='4xx'),
+        stats.Observer(key='downloader/response_status_count/5..', use_re_key=True, export_key='5xx'),
+    ]
+
+    """
+    Generated stats:
+    {'2xx': 10}                               # for t=1
+    {'2xx': 34, '3xx': 1, '4xx':1}            # for t=2
+    {'2xx': 67, '3xx': 4, '4xx':2, '5xx': 1}  # for t=3
+    {'2xx': 90, '3xx': 8, '4xx':2, '5xx': 1}  # for t=4
+    ...
+    """
+
+.. _topics-stats-statspipelines:
+
+Stats Pipelines
+===============
+
+:class:`PeriodicStatsPipeline` objects receive and process the stats generated by the :class:`PeriodicStatsObserver`
+objects for every :ref:`topics-stats-periodicstatscollector` iteration.
+
+You can define the pipeline objects to use with the :setting:`PERIODIC_STATS_PIPELINES` setting::
+
+    PERIODIC_STATS_PIPELINES = [
+        'scrapy.contrib.periodicstats.PeriodicStatsLogger',
+    ]
+
+
+PeriodicStatsPipeline
+---------------------
+
+.. module:: scrapy.contrib.periodicstats
+.. class:: PeriodicStatsPipeline
+
+    .. method:: open_spider(spider)
+
+        Called when the spider is opened.
+
+        :param spider: the spider for which the stats are processed.
+        :type spider: :class:`~scrapy.spider.Spider` object
+
+    .. method:: close_spider(spider)
+
+        Called when the spider is closed.
+
+        :param spider: the spider for which the stats are processed.
+        :type spider: :class:`~scrapy.spider.Spider` object
+
+    .. method:: process_stats(spider, interval, stats)
+
+        Called for every :class:`PeriodicStatsCollector` iteration with the stats to be processed.
+
+        :meth:`process_stats` should return the passed stats or ``None``.
+
+        If it returns the passed stats, they will be passed to the next middleware.
+
+        If it returns ``None``, no further middlewares will be called.
+
+        :param spider: the spider for which the stats are processed.
+        :type spider: :class:`~scrapy.spider.Spider` object
+
+
+You can define your own :class:`PeriodicStatsPipeline` objects to periodically export stats.
```

Comment:
One problem here is that we don't help users with "export" part. Export likely means "send data via HTTP", and it is not clear how to do it in a non-blocking way. Most users will just use requests.